src/System.Composition.Convention/tests/ExceptionAssert.cs

```diff
@@ -10,315 +10,31 @@ namespace System.Composition.Convention
-                AssertSameParameterName(parameterName, actual, retryCount);
-            });
-
-            return exception;
-        }
-
-        /// <summary>
-        ///     Verifies that the specified action throws an exception of type <typeparam name="T"/>,
-        ///     with the specified inner exception of type <typeparam name="TInner"/>.
-        /// </summary>
-        public static T Throws<T, TInner>(Action action)
-            where T : Exception
-            where TInner : Exception
-        {
-            return Throws<T, TInner>(RetryMode.Retry, action);
-        }
-
-        /// <summary>
-        ///     Verifies that the specified action throws an exception of type <typeparam name="T"/>,
-        ///     with the specified inner exception of type <typeparam name="TInner"/>, and indicating 
-        ///     whether to retry.
-        /// </summary>
-        public static T Throws<T, TInner>(RetryMode retry, Action action)
-            where T : Exception
-            where TInner : Exception
-        {
-            return Throws<T, TInner>(retry, action, (Action<T, int>)null);
-        }
-
-        /// <summary>
-        ///     Verifies that the specified action throws an exception of type <typeparam name="T"/>,
-        ///     with the specified inner exception of type <typeparam name="TInner"/>, indicating 
-        ///     whether to retry and running the specified validator.
-        /// </summary>
-        public static T Throws<T, TInner>(RetryMode retry, Action action, Action<T, int> validator)
-            where T : Exception
-            where TInner : Exception
-        {
-            var exception = Throws<T>(retry, action, (actual, retryCount) =>
-            {
-                AssertIsExactInstanceOfInner(typeof(TInner), actual, retryCount);
-
-                if (validator != null)
-                {
-                    validator(actual, retryCount);
-                }
-            });
-
-            return exception;
-        }
-
-        /// <summary>
-        ///     Verifies that the specified action throws an exception of type <typeparam name="T"/>,
-        ///     with the specified inner exception.
-        /// </summary>
-        public static T Throws<T>(Exception innerException, Action action)
-            where T : Exception
-        {
-            return Throws<T>(innerException, RetryMode.Retry, action, (Action<T, int>)null);
-        }
-
-        /// <summary>
-        ///     Verifies that the specified action throws an exception of type <typeparam name="T"/>,
-        ///     with the specified inner exception, and indicating whether to retry.
-        /// </summary>
-        public static T Throws<T>(Exception innerException, RetryMode retry, Action action)
-            where T : Exception
-        {
-            return Throws<T>(innerException, RetryMode.Retry, action, (Action<T, int>)null);
-        }
+        // NOTE: To catch state corrupting exceptions, it by design that
```

Comment:
Nit: "it is by design" (I realize the error was there in the original, just pointing it out)